Offer to add local user to pihole group on fresh installations

yubiuser opened the pull request:
**What does this PR aim to accomplish?:**

Certain CLI functions e.g. `pihole -v` or `pihole api` require access to files in `/etc/pihole` which belong to `pihole:pihole`. This PR adds a dialog to offer to put the user with UID 1000 (usually the first local user) into the `pihole` group during installation.

To be discussed.

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

yubiuser commented:
Inspired by https://github.com/pi-hole/pi-hole/pull/6002#issuecomment-2779961094

mwoolweaver commented:
instead of adding id `1000` could something like what is mentioned [here](https://askubuntu.com/a/1034050)? 

`$SUDO_USER` could be used to get the user that called the script with sudo.

dschaper commented:
I'm not comfortable with this. I can't think of any other package that offers this, I know Docker (get.docker.com) prints out a message to advise users to add the proper user to the right group(s) but I don't like the idea of doing it programmatically. Too many ways this could go very wrong.

mwoolweaver commented:
> I can't think of any other package that offers this

I only know of [one](https://wiki.servarr.com/install-script).

yubiuser commented:
> prints out a message to advise users to add the proper user to the right group(s) 

We could also reduce the scope of this PR to only show an advise to do this.

PromoFaux commented:
In my opinion we should just set versions file consistently to `644` (so, service wrapper needs updating) and not worry about this. 

There is far too much complexity added in this PR for extremely little gain.

> We could also reduce the scope of this PR to only show an advise to do this.

If anything, then this - but with no checks for the user ID or anything like that. docs.pi-hole.net probably a good target for somewhere to have this information.

dschaper commented:
We could tag on a note to the last dialog displayed that shows the generated password. Something like https://github.com/docker/docker-install/blob/master/install.sh#L289. It looks like the verbiage has been updated to point to the documentation now instead of suggesting adding the user to the docker group. https://github.com/docker/docker-install/commit/b5567e71b7adca1f42db82b7f9bcd6fb36909bcb

![image](https://github.com/user-attachments/assets/85b02cee-9d74-481c-82f9-4bcd120c0a95)

yubiuser commented:
I removed to code for automatic addition and added some info to the final dialog.
![2025-04-07_14-58](https://github.com/user-attachments/assets/65c9e257-6743-44bf-9046-086ccf1763e1)

PromoFaux commented on automated install/basic-install.sh:
```diff
@@ -2489,7 +2489,13 @@ main() {
+\\n
+\\n
+\\nTo allow your user to use all CLI functions without authentication,\
+\\nadd your user to the 'pihole' group with something like:\
```
```suggestion
\\nadd your user to the 'pihole' group with:\
```

"Something like" is a bit too non-specific.

Also, there is some similar output to the CLI - possibly worth putting it there, too.

yubiuser replied:
Done.

dschaper commented:
Can we see the dialog with the final changes made please?

yubiuser commented:
![2025-05-11_20-42](https://github.com/user-attachments/assets/c0aea06a-7430-461e-8e36-7418ce0adb43)

dschaper commented:
Thanks, I can tell you now that we will get support requests saying `sudo usermod -aG pihole $your_user` failed. 

But I'll re-request Promo on it.

yubiuser commented:
I know... I think you suggested to do it like `docker` which used the phrase above. I added `something like` because it might be a bit different syntax on different OS. But Promo complaint....

We can also get rid of an example command, and only leave 

> add your user to the 'pihole' group.

But this will invoke even more follow-up questions....

PromoFaux commented:
> We can also get rid of an example command, and only leave

To be honest, I think this is probably best. how about a section on the docs site that shows a different examples for different OS, and provide a link in the dialog rather than an example command.

dschaper commented:
> But this will invoke even more follow-up questions....

That's exactly how the docker flow is now, 
```bash
	echo
	echo "To run the Docker daemon as a fully privileged service, but granting non-root"
	echo "users access, refer to https://docs.docker.com/go/daemon-access/"
	echo
	echo "WARNING: Access to the remote API on a privileged Docker daemon is equivalent"
	echo "         to root access on the host. Refer to the 'Docker daemon attack surface'"
	echo "         documentation for details: https://docs.docker.com/go/attack-surface/"
	echo
	echo "================================================================================"
	echo
```

yubiuser commented:
I changed the dialog/cli to refer to our post-installation documentation and squashed the commits. 

The doc PR is at https://github.com/pi-hole/docs/pull/1244

![2025-05-14_09-34](https://github.com/user-attachments/assets/f42280e9-b66e-4764-a703-3b1281f4f087)
